File: code/csharp/DesignPatterns/prototype/PrototypeClient.cs
using System;

namespace Prototype
{
    public class PrototypeClient
    {
        public static void Prototype()
        {
            IPrototype teddyBear1 = new WhiteTeddyBearPrototype();
            IPrototype teddyBear2 = new BlackTeddyBearPrototype();

            IPrototype clonedTeddyBear1 = teddyBear1.Clone();
            IPrototype clonedTeddyBear2 = teddyBear2.Clone();

            Console.WriteLine($"Cloned Teddy Bear 1: {clonedTeddyBear1.GetType().Name}");
            Console.WriteLine($"Cloned Teddy Bear 2: {clonedTeddyBear2.GetType().Name}");
        }
    }
}

File: code/typescript/prototype/PrototypeClient.ts
import { BlackTeddyBearPrototype } from "./BlackTeddyBearPrototype";
import { WhiteTeddyBearPrototype } from "./WhiteTeddyBearPrototype";

export function Prototype(): void {
    let teddyBear1 = new WhiteTeddyBearPrototype();
    let teddyBear2 = new BlackTeddyBearPrototype();

    let clonedTeddyBear1 = teddyBear1.clone();
    let clonedTeddyBear2 = teddyBear2.clone();

    console.log("Cloned Teddy Bear 1:", clonedTeddyBear1.constructor.name);
    console.log("Cloned Teddy Bear 2:", clonedTeddyBear2.constructor.name);
}
